addons/project/project.py

```diff
@@ -825,6 +825,19 @@ def message_get_suggested_recipients(self, cr, uid, ids, context=None):
                 data._message_add_suggested_recipient(recipients, partner=data.partner_id, reason=reason)
         return recipients
 
+    @api.multi
+    def message_get_email_values(self, notif_mail=None):
+        self.ensure_one()
+        res = super(task, self).message_get_email_values(notif_mail=notif_mail)
+        headers = res.get('headers', {})
+        if headers.get('X-Odoo-Objects') and self.project_id:
+            project_info = 'project.project-%s, ' % (self.project_id.id)
+            headers['X-Odoo-Objects'] = project_info + headers['X-Odoo-Objects']
```

Comment:
This is not coherent with the X-Odoo-Objects meaning. Sometimes the record, sometimes its "parent". Should be stored in separate parameters.